### PR TITLE
Add check for version guards in erb templates

### PR DESCRIPTION
**Pull request by roaks3**
We sometimes break downstream providers when invalid version guards are introduced in .erb files. This happens because we have 3+ versions to consider, with each being a superset of the next:

"private" or "alpha" -> "beta" -> "ga"

The typical version guard would be `<% unless version == 'ga' -%>`, which means the feature should not be available in the GA provider, but should be in Beta and Private/Alpha. Sometimes users will assume `<% if version == 'beta' -%>` is equivalent, but it means the feature will NOT be available in Private/Alpha, which is almost always wrong.

In short, our features should never target only beta, because it would break the assumption of Private/Alpha being a superset. This logic also works in reverse (where we wouldn't want any GA-specific functionality to also be present in Private/Alpha).

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by roaks3** [comment]
@c2thorn this isn't complete yet, but I was wondering if you could give it a quick look to make sure the approach seems reasonable, and will not be too much work to change later (assuming we ultimately stop using erb templates). I'm planning to have it run with the other mm checks against all erb files that have changed in the submitted diff.

**Comment by c2thorn** [comment]
This would run as a separate check on PR's?

Yeah I think this is a good approach. Seems easy enough to swap out the identifying text when we swap to Go `.tmpl` templates. It will eventually look something along the lines of `{{- if eq $.MinVersion "beta" }}`

**Comment by roaks3** [comment]
Great, thanks!

Yea, it would be a separate PR check, something to the effect of `git diff --name-only ../../*.erb | go run main.go`

**Comment by roaks3** [comment]
Note: this is ready for review now. The plan is to run this command in a check as a separate PR.

**Review by c2thorn — APPROVED** [review]
